### ISSUE TITLE
Configuration for Blazor WebAssembly

### ISSUE THREAD

**Issue by danroth27** [dotnet/aspnetcore#18675]
We provide the config abstraction in the Blazor WebAssembly host, but we don't currently populate it with anything. We should consider what sort of out of the box support we want for configuration in Blazor WebAssembly apps.

**Comment by MarkFl12** [comment]
I have my WebAssembly project hosted in an ASP.NET core site that's running in docker. I'd like to be able to pass environment variables to the docker container and have them affect the configuration of the WebAssembly on the client.

**Comment by pranavkm** [comment]
```
// Blazor.Build
<!-- Do not publish the Development.json -->
<ItemGroup>
    <Content Update="wwwroot/appsettings.Development.json" CopyToPublishDirectory="Never" />
</ItemGroup>
```

```
Blazor.Boot.json will have an additional section that includes all available appsettings.* file. This helps avoid 404s

// boot.json
{
    "config": [
        "appSettings.Development.json",
        "appSettings.json"
    ]
}
```
```
// We have a middleware that adds a header for requests to blazor.boot.json

// Middleware - DevServer + Hosted template
if (app.IsDevelopment())
{
    app.Use(async (context, response) =>
    {
       response.Headers.Add("ComponentsWebAssemblyEnv", "Development"); 
    });
}
```

```ts
// Blazor.WebAssembly.ts

const response = await fetch('blazor.boot.json');
// Make it visible to user code?
const env = response.headers["ComponentsWebAssemblyEnv"] ?? 'Production';

if (boot.config) {
    // boot manifest contains filtered list
    boot.config.forEach(fetch);
    
    
    // boot manifest contains all files, but we filter on the client
    if (boot.config.contains("appsettings.json"))
        fetch('appsettings.json');
    
    if (env && boot.config.contains(`appsettings.${env}.json`)) {
        fetch(`appsettings.${env}.json`);
    
}
```

Open questions:

* Do we expose the environment via `IWebAssemblyHost` i.e. `IWebAssemblyHost.IsDevelopment`?

**Comment by knight1219** [comment]
I created [this package](https://github.com/Darixidor/Blazor.Settings) because I needed access to the environment in a ASP.Net Core hosted situation. In a CDN/plain file hosted situation I'm not sure how you could get the client to understand the variables.

**Comment by arivera12** [comment]
How to set the host configuration?

host.Configuration = ?

Configuration is only a get property there is not set method inside the WebAssemblyHost.

My app is entirely broken by this change. 

How to set up the configuration?

**Comment by pranavkm** [comment]
**Hi, it looks like you are posting on a closed issue/PR/commit!**

We're very likely to lose track of your bug/feedback/question unless you:
1. Open a new issue
2. Explain very clearly what you need help with
3. If you think you have found a bug, include detailed repro steps so that we can investigate the problem

Thanks!

**Comment by arivera12** [comment]
I just resolved the issue. 

There are other people asking for this and there is no example or documentation on this.

I will left here the solution after 2 wasted days crashing on all errors and working around with this.

This
`WebAssemblyHostBuilder builder = WebAssemblyHostBuilder.CreateDefault(args);`

Holds the configuration builder which enables to insert a configuration file.

`//Example`
`builder.Configuration.AddJsonFile(//your config file were ever its located or loaded)`

Take note that DON'T BUILD THE CONFIGURATION file it will throw an error:

```
//Don't do this
builder.Configuration.Build();
```

Let

`WebAssemblyHost host = builder.Build();`

Build the configuration file for you.

Then the configuration file now will be loaded and available from

`var configuration = host.Services.GetRequiredService<IConfiguration>();`

Or

```
[Inject]
 IConfiguration Configuration { get; set; }
```

**Comment by razfriman** [comment]
@danroth27 Are there any plans to support loading a Configuration from the server project hosting the Blazor WASM?

Use Case: Passing in the API-url to the project.

This tends to happen via environment variables in that get passed in through Docker and are not known at the time of writing the application.

**Comment by danroth27** [comment]
Hi @razfriman. We now have support in Blazor WebAssembly for loading environment specific configuration from the server: https://docs.microsoft.com/aspnet/core/blazor/hosting-model-configuration#configuration

**Comment by razfriman** [comment]
This 

> Hi @razfriman. We now have support in Blazor WebAssembly for loading environment specific configuration from the server: https://docs.microsoft.com/aspnet/core/blazor/hosting-model-configuration#configuration

Thank you!!

That is exactly what I needed to unblock the switch over to Blazor WebAssembly branch for my app. 

Thank you x a million to the whole team
